tools/as/as.cpp

```diff
@@ -129,7 +129,7 @@ int main(int argc, char** argv) {
   }
 
   std::vector<char> contents;
-  if (!ReadFile<char>(inFile, "r", &contents)) return 1;
+  if (!ReadFile<char>(inFile, /* isBinaryFile = */ false, &contents)) return 1;
```

Comment:
nit:  The temptation to use a comment on the parameter expression is a hint that two methods would be easier to use:  ReadBinaryFile and ReadTextFile.

But this MR is a vast improvement over the current situation.  So I won't block on this.

Reply:
good idea, and I need to fix the PR anyway, so I'll do that.